src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -6691,8 +6699,7 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
         // If tmp is a statement and not an expression
         // never cast into expression using ASRUtils::EXPR
         // Just ignore and exit the function naturally.
-        if( tmp && !ASR::is_a<ASR::stmt_t>(*tmp) ) {
-            LCOMPILERS_ASSERT(ASR::is_a<ASR::expr_t>(*tmp));
+        if( tmp && !ASR::is_a<ASR::expr_t>(*tmp) ) {
```

Comment:
This is changed to keep track of the final expression executed, and printing its result.

Reply:
I didn't get this. Could you elaborate this, please?

Reply:
Please look at the difference in the AST generated with this change and without this change (i.e. this PR and the main branch). It may become easier to understand. Start the interactive shell with `-v` option to print the AST and LLVM IR.

With this change, the `pass_wrap_global_stmt` will generate a function that returns the result of the expression `tmp`, avoiding the modifications you specified with `make_dummy_assignment` (https://github.com/lcompilers/lpython/pull/2698#pullrequestreview-2051417737).